docs/source/diagnostics-performance.rst

```diff
@@ -0,0 +1,16 @@
+Diagnostic and Performance
```

Comment:
We should consider adding an "s" to the end of Diagnostic

Altenatively, here are some names:

- Understanding Performance
- Debugging and Performance
- Troubleshooting and Performance

Diagnostic sounds a little clinical to me.  I think that terms like Debugging and Troubleshooting are more likely what users will look for.